src/components/forms/DateField/DateInputGroup.js

```diff
@@ -141,10 +116,22 @@ const DateInputGroup = ({
   });
 
   const enteredDate = dateFromParts(dateParts.year, dateParts.month, dateParts.day);
-  useEffect(() => {
-    if (!enteredDate || enteredDate === value) return;
-    onChange({target: {name, value: enteredDate}});
-  }, [onChange, name, value, enteredDate]);
+
+  const onPartChange = event => {
+    const {
+      target: {name: partName, value},
+    } = event;
+    const newDateParts = {...dateParts, [partName]: value};
+    // update internal state
+    setDateParts(newDateParts);
+
+    // calculate the nw formik state value for the "composite" field
+    const {year, month, day} = newDateParts;
+    const newDate = dateFromParts(year, month, day);
+
+    // clear value if it's not a valid date or update it if it is valid
+    onChange({target: {name, value: newDate ?? ''}});
```

Comment:
I don't think so, but I'm not the biggest advocate for this pattern either. I tried overwriting the source event via:

```js
{...event, target: {...event.target, value: newDate}}
```
but that broke more because `event` is really a React `SyntheticEvent` and not a plain object, which caused it to lose the `name` property along the way. And then I gave up and simplified things.

I couldn't find any resources either about how you'd modify the original event with a processed value :(